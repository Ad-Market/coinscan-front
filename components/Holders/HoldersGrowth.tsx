import Box from '@mui/material/Box';
import Skeleton from '@mui/material/Skeleton';

import GrowthPercent from './HoldersGrowthPercent';

type GrowthProps = {
  holdersCount: number;
  oneday: number;
  sevendays: number;
  thirtydays: number;
  loading?: boolean;
};

const Growth = ({
  holdersCount,
  oneday,
  sevendays,
  thirtydays,
  loading,
}: GrowthProps) => {
  return (
    <Box>
      <Box sx={{ color: 'text.primary' }}>Total Holders</Box>
      <Box
        sx={{
          color: 'text.primary',
          fontSize: '3rem',
          fontWeight: 'medium',
        }}
      >
        {loading ? (
          <Skeleton sx={{ height: 70, width: { xs: 80, sm: 150 } }} />
        ) : (
          holdersCount && holdersCount.toLocaleString('en-US')
        )}
      </Box>
      <Box
        sx={{
          display: 'flex',
          flexDirection: { xs: 'row', sm: 'column' },
        }}
      >
        <GrowthPercent loading={loading} value={oneday} text="1d" />
        <GrowthPercent loading={loading} value={sevendays} text="7d" />
        <GrowthPercent loading={loading} value={thirtydays} text="30d" />
      </Box>
    </Box>
  );
};

export default Growth;
