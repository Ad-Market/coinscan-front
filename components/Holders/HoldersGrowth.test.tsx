import { render } from '@testing-library/react';
import HoldersGrowth from './HoldersGrowth';

test('loading state', () => {
  const { container } = render(
    <HoldersGrowth
      holdersCount={100}
      oneday={10}
      sevendays={30}
      thirtydays={40}
      loading
    />
  );

  expect(container).toMatchSnapshot('loading state');
});
