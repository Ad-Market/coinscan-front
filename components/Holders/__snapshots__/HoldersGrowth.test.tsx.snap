// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading state: loading state 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiBox-root css-1jolfkj"
    >
      Total Holders
    </div>
    <div
      class="MuiBox-root css-1n4i7xh"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-bsas7j-MuiSkeleton-root"
      />
    </div>
    <div
      class="MuiBox-root css-51kbo7"
    >
      <div
        class="MuiBox-root css-9od62j"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-p48zi-MuiSkeleton-root"
        />
      </div>
      <div
        class="MuiBox-root css-9od62j"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-p48zi-MuiSkeleton-root"
        />
      </div>
      <div
        class="MuiBox-root css-9od62j"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-p48zi-MuiSkeleton-root"
        />
      </div>
    </div>
  </div>
</div>
`;
